flask_resources/resources.py

```diff
@@ -24,17 +25,17 @@
 class ResourceConfig:
     """Base resource configuration."""
 
-    item_request_loaders = {
-        "application/json": JSONLoader(),
-        "application/json+patch": JSONPatchLoader(),
+    request_loaders = {
+        "application/json": RequestLoader(
+            deserializer=JSONDeserializer(),
+            item_args_parser=item_request_parser,
+            search_args_parser=search_request_parser,
+        )
```

Comment:
1- First of all the `item_request_parser` got removed since it was parsing the "id" which is actually a part of the URL not a parameter. The `create_request_parser` also got removed, since it was not parsing any field, which is the same behaviour than actually not parsing 😅 .

In terms of hierarchy. I agree that is more general than a MIMEType, however is the `RequestLoader` who needs it, not the MIMEType... Therefore, they would be re-usable (i.e. different loaders can have the same args_parse).

About naming I opened <INSERT ISSUE LINK> I think is a larger topic and would block it... A quick thought. I am not a fan of inverting loader/deserializers. Mainly because we haver serializer on the other side, then it the counter word (serializer --> deserializer) would mean different things depending on input/output direction. About loaders/response I am open to suggestions :)

Reply:
For 1- it''s mostly strange to me to tie a RequestLoader to a specific mimetype but a generic querystring parsing forcing the developer to repeat themselves an keep things in sync... but yes let's do more rounds of using to see.